### PR TITLE
*: adaptive analyze scan concurrency

### PR DESCRIPTION
**Pull request by hawkingrei**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17687

- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1118,7 +1118,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Type: Integer
 - Default value: `4`
-- Range: `[1, 4294967295]`
+- Range: `[0, 4294967295]`. In versions prior to v8.2, the minimum value was 1. When set to 0, it will adapt the concurrency based on the cluster size.
```
Comment (hfxsd) [inline system-variables.md:1121]:
```suggestion
- Range: `[0, 4294967295]`. In versions prior to v8.2.0, the minimum value is `1`. When you set it to `0`, it adapts the concurrency based on the cluster size.
```

File: system-variables.md [hunk 2]
```diff
@@ -5620,7 +5620,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Type: Integer
 - Default value: `1`
-- Range: `[1, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`.
+- Range: `[0, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`. In versions prior to v8.2, the minimum value was 1. When set to 0, it will adapt the concurrency based on the cluster size.
```
Comment (hfxsd) [inline system-variables.md:5623]:
```suggestion
- Range: `[0, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`. In versions prior to v8.2.0, the minimum value is `1`. When you set it to `0`, it adapts the concurrency based on the cluster size.
```

**Review by songrijie — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -5620,7 +5620,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Type: Integer
 - Default value: `1`
-- Range: `[1, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`.
+- Range: `[0, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`. In versions prior to v8.2.0, the minimum value is `1`. When you set it to `0`, it adapts the concurrency based on the cluster size.
```
Comment (hfxsd) [inline system-variables.md:5623]:
```suggestion
- Range: `[0, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`. Before v8.2.0, the minimum value is `1`. When you set it to `0`, it adapts the concurrency based on the cluster size.
```

File: system-variables.md [hunk 4]
```diff
@@ -1118,7 +1118,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Type: Integer
 - Default value: `4`
-- Range: `[1, 4294967295]`
+- Range: `[0, 4294967295]`. In versions prior to v8.2.0, the minimum value is `1`. When you set it to `0`, it adapts the concurrency based on the cluster size.
```
Comment (lilin90) [inline system-variables.md:1121]:
```suggestion
- Range: `[0, 4294967295]`. In versions earlier than v8.2.0, the minimum value is `1`. When you set it to `0`, it adaptively adjusts the concurrency based on the cluster size.
```

File: system-variables.md [hunk 5]
```diff
@@ -5620,7 +5620,7 @@ For details, see [Identify Slow Queries](/identify-slow-queries.md).
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
 - Type: Integer
 - Default value: `1`
-- Range: `[1, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`.
+- Range: `[0, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`. Before v8.2.0, the minimum value is `1`. When you set it to `0`, it adapts the concurrency based on the cluster size.
```
Comment (lilin90) [inline system-variables.md:5623]:
```suggestion
- Range: `[0, 4294967295]`. The maximum value for v7.5.0 and earlier versions is `256`. Before v8.2.0, the minimum value is `1`. When you set it to `0`, it adaptively adjusts the concurrency based on the cluster size.
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**